From 77879838383f0ab86f3c19938f05444d8f6e1c38 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Sun, 13 Nov 2016 09:24:15 -0800
Subject: [PATCH] DispSync: Use boottime to fix timestamp mismatch

Monotonic time does not include suspend time, causing the expected
timestamps to be skewed. This results in the following being spammed to
the log:
SurfaceFlinger: [SurfaceView] Timestamp 2913013563839 seems implausible
relative to expectedPresent 1640907033276

Use boottime instead of monotonic to fix the timestamp skew.

Change-Id: Ie5ef2dd778c514a16ecf000580dabefbecedb800
---
 services/surfaceflinger/DispSync.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/surfaceflinger/DispSync.cpp b/services/surfaceflinger/DispSync.cpp
index 192d746..e7b3c4a 100644
--- a/services/surfaceflinger/DispSync.cpp
+++ b/services/surfaceflinger/DispSync.cpp
@@ -518,7 +518,7 @@ void DispSync::resetErrorLocked() {
 
 nsecs_t DispSync::computeNextRefresh(int periodOffset) const {
     Mutex::Autolock lock(mMutex);
-    nsecs_t now = systemTime(SYSTEM_TIME_MONOTONIC);
+    nsecs_t now = systemTime(SYSTEM_TIME_BOOTTIME);
     nsecs_t phase = mReferenceTime + mPhase;
     return (((now - phase) / mPeriod) + periodOffset + 1) * mPeriod + phase;
 }
-- 
2.9.3

